Add PGPASS User Manager

pgwijesinghe opened the pull request:
@ciozi137 I pushed the changes we discussed at https://github.com/levylabpitt/LV-Data/issues/102 to this new branch "102-pg-pgpass-gui". You can review it here.

pgwijesinghe committed:
Add PGPASS User Manager

ciozi137 committed:
Updates to PGPASS library:
- UserMgr: refactoring into jki state machine (incomplete)
- UserMgr: experiment with listbox vs ring
- Write: handle deduping connections and some refactoring
- Select Connection: call UserMgr if there is an issue with the connection string (note: should also test connection?)

pgwijesinghe committed:
Modify PGPASS User Manager

-JKI State Machine Refactoring
-UI Changes
-Added deduping logic
-PGPASS/Write: Removed deduping implementation (added it as a state in UserMgr)

pgwijesinghe commented:
@ciozi137 I modified the VI and it's open to review. Following is a summary of the changes:
- Carried on with the State Machine implementation
- Changed the ring to a list box as you suggested.
- Implemented deduping logic as a state
  - When adding a new user, If, host AND dbname AND username == 1 -> dedupe. Else, proceed. I think this satisfies our requirement. let me know if there's an issue with this logic.
- Add User, Delete User functionality
- Login state is still incomplete as I'm unsure of what to output.

ciozi137 committed:
Updates to PGPASS library
- user --> connection/s
- granularize states
- update UI

ciozi137 commented:
@pgwijesinghe 10 minute update: https://drive.google.com/file/d/1wgCN5VelIn57nXdvIqgyiriFBKUozzUA/view?usp=drivesdk

ciozi137 committed:
refining code re: Issue #102

ciozi137 committed:
changes related to handling connections to the database:
- Issue #100: handle save/load previous known good connection string
- Issue #102: further improvements to connection manager

ciozi137 committed:
remove unused PGPASS.dialog

ciozi137 committed:
Issue #105:
- update Get Trace and Get Traces query with "AND nullif(%s, 'NaN') IS NOT NULL" clause

ciozi137 committed:
Issue #101:
- process multiline insert in 1000 row chunks
- add new error for data array not equal size of timestamp array

ciozi137 committed:
Issue #104:
- verify switch tables
